tutorials/v7/draw_mt.cxx

```diff
@@ -91,7 +91,7 @@ void draw_mt()
    std::thread thrd2(draw_canvas, "Second canvas", RColor::kBlue);
    std::thread thrd3(draw_canvas, "Third canvas", RColor::kGreen);
 
-   thrd1.detach();
-   thrd2.detach();
-   thrd3.detach();
+   thrd1.join();
+   thrd2.join();
+   thrd3.join();
 }
```

Comment:
If I (or the test suite) runs with `root -q` we call `terminate` which is ever worse than not having the prompt. We might need some thread registry for interactivity, where upon global destruction threads are `join`ed.

Reply:
Yes, we missed this discussion completely during workshop.

One need different utilities to handle "interactive" threads, where user code plus some service ROOT functions must be executed. But simple join() does not help - such threads need to have some global termination condition on which they should react

Simple solution for now - enable join() for batch mode

Reply:
We cannot crash interactively if people `.q` too early. Tutorials need to work, properly. It's also a really bad example to `detach()` a thread and then just "leave". A tutorial is user-facing code and must be shiny and correct, without hacks. We can get rid of the `join()`s once we have the thread registration mechanism in ROOT.

Reply:
> We cannot crash interactively if people .q too early. Tutorials need to work, properly. 

I agree. Since we do not have safe method to stop such threads - let use join.
I can add relevant parts of these changes into #2603 - anyway I need to adjust it now.